### PR TITLE
Update development build for master branch

### PR DESCRIPTION
**Pull request by guozhangwang**
@miguno @enothereska @bbejeck @dguy @mjsax 

Note that `master` points to `3.5` not `3.4`.

### CODE REVIEW

File: kafka-streams/README.md [hunk 1]
```diff
@@ -185,77 +185,34 @@ repository may have different Kafka requirements.
...
-  "org.apache.kafka" % "kafka-clients" % "0.10.2.1-cp1"
-)
-```
-
 <a name="requirements-confluent-platform"/>
 
 ## Confluent Platform
 
-The code in this repository requires Confluent Platform 3.1.x.
+The code in this repository requires [schema-registry](https://github.com/confluentinc/schema-registry) from the Confluent Platform.
```
Comment (miguno) [inline kafka-streams/README.md]: Nit: requires [Confluent] schema registry.

File: kafka-streams/README.md [hunk 2]
```diff
@@ -185,77 +185,34 @@ repository may have different Kafka requirements.
...
 * [Confluent Platform Quickstart](http://docs.confluent.io/current/quickstart.html) (how to download and install)
 * [Confluent Platform documentation](http://docs.confluent.io/current/)
 
-If you just run the integration tests (`mvn test`), then you do not need to install anything -- all maven artifacts
-will be downloaded automatically for the build.  However, if you want to interactively test-drive the examples under
-[src/main/](src/main/)
-(such as [WordCountLambdaExample](src/main/java/io/confluent/examples/streams/WordCountLambdaExample.java)), then you
-do need to install Confluent Platform.  See [Packaging and running the examples](#packaging-and-running) below.  Also,
-each example states its exact requirements at the very top.
+> **For the `master` branch:** To build a development version, you typically need the latest `master` version of Confluent Platform's
```
Comment (miguno) [inline kafka-streams/README.md]: Nit: I'd always name it "Confluent schema registry" (or even "Confluent Schema Registry"), not "schema-registry".

File: kafka-streams/README.md [hunk 3]
```diff
@@ -185,77 +185,34 @@ repository may have different Kafka requirements.
...
+> Schema Registry (cf. `confluent.version` in [pom.xml](pom.xml) for details). The following instructions will build and locally install
+> the latest `master` Schema Registry version:
+>
+> ```shell
+> $ git clone https://github.com/confluentinc/schema-registry.git
```
Comment (miguno) [inline kafka-streams/README.md]: This is not sufficient.  You must also build e.g. confluentinc/common and confluentinc/rest-tuils!

File: kafka-streams/README.md (hunk 2 above)
Comment (guozhangwang) [inline kafka-streams/README.md]: ack.

File: kafka-streams/README.md (hunk 3 above)
Comment (guozhangwang) [inline kafka-streams/README.md]: I already mentioned that in the doc, but maybe it is better to add in the shell example as well.

**Pull request merged**